ci-operator/step-registry/ipi/conf/azure/confidential/ipi-conf-azure-confidential-commands.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [[ -z "${SECURITY_TYPE}" ]];  then
+    echo "ENV SECURITY_TYPE is not set, exit."
+    exit 1
+fi
+
+echo "SECURITY_TYPE: ${SECURITY_TYPE}"
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+PATCH="/tmp/install-config-security.yaml.patch"
+if [[  "${SECURITY_TYPE}" == "TrustedLaunch" ]]; then
+cat > "${PATCH}" << EOF
+controlPlane:
+  platform:
+    azure:
+      encryptionAtHost: true
+      settings:
+        securityType: TrustedLaunch
+        trustedLaunch:
+          uefiSettings:
+            secureBoot: Enabled
+            virtualizedTrustedPlatformModule: Enabled
+compute:
+- platform:
+    azure:
+      encryptionAtHost: true
+      settings:
+        securityType: TrustedLaunch
+        trustedLaunch:
+          uefiSettings:
+            secureBoot: Enabled
+            virtualizedTrustedPlatformModule: Enabled
+EOF
+fi
+
+if [[  "${SECURITY_TYPE}" == "ConfidentialVM" ]]; then
+    if [[ -z "${SECURITY_ENCRYPTION_TYPE}" ]]; then
+        echo "securityEncryptionType must be set when security type is ConfidentialVM."
+        exit 1
+    fi
+
+cat > "${PATCH}" << EOF
+platform:
+  azure:
+    defaultMachinePlatform:
+      encryptionAtHost: true
+      settings:
+        securityType: ConfidentialVM
+        ConfidentialVM:
+          uefiSettings:
+            secureBoot: Enabled
+            virtualizedTrustedPlatformModule: Enabled
+      osDisk:
+        securityProfile:
+          securityEncryptionType: ${SECURITY_ENCRYPTION_TYPE}
+EOF
+    if [[ "${SECURITY_ENCRYPTION_TYPE}" == "DiskWithVMGuestState" ]] && [ -f "${provisioned_des_file}" ]; then
+        provisioned_rg_file="${SHARED_DIR}/resourcegroup"
```

Comment:
des is a must for `DiskWithVMGuestState` type?

Reply:
no, when using DiskWithVMGuestState type, vm can use platform-managed keys (PMK) or a customer-managed key (CMK) to encrypt os disk. If we defines DES, customer-managed key will be used, else platform-managed keys will be used by default.

Discussed with @miyadav before, we plan to choose customer-managed key for confidentialVM + DiskWithVMGuestState for CI integration when it can be tested, to make sure that cpms and machineset works well.

Reply:
got it.